frodo.js

```diff
@@ -1,64 +1,66 @@
-/**
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.